# Exercício 18 (MINI BOSS): Análise de Desempenho Escolar
## Você é um cientista de dados contratado por uma escola para analisar o desempenho de três turmas (Turma A, Turma B, Turma C) em duas disciplinas: Matemática e Ciências.
### O objetivo é gerar um relatório completo utilizando as medidas estatísticas estudadas para entender o desempenho geral dos alunos e descobrir padrões importantes.


# Tarefas:


## 1. Carregar o dataset que contém as notas dos alunos das três turmas. O arquivo chama 18_analise_avancada.csv.

In [3]:
import pandas as pd

analise_avancada = pd.read_csv('18_analise_avancada.csv')
analise_avancada

Turma  Matemática  Ciências
0      C          83        96
1      A          47        91
2      C          63        52
3      C          50        80
4      A          90        42
..   ...         ...       ...
95     A          78        79
96     A          72        78
97     C          40        57
98     A          96        79
99     A          66        40

[100 rows x 3 columns]

## 2. Calcular as seguintes medidas estatísticas para cada turma (A, B e C) em Matemática e Ciências:

a. Média\
b. Mediana\
c. Moda\
d. Amplitude (diferença entre a maior e a menor nota)\
e. Variância\
f. Desvio Padrão\
g. Covariância e Correlação entre Matemática e Ciências

In [14]:
# Filtrando o DF

turma_a = analise_avancada[analise_avancada['Turma'] == 'A']
turma_b = analise_avancada[analise_avancada['Turma'] == 'B']
turma_c = analise_avancada[analise_avancada['Turma'] == 'C']


### a) Média:

In [18]:
# Turma A
media_a = turma_a[['Matemática', 'Ciências']].mean()

mensagem = f'''Média das notas:
Matemática: {media_a['Matemática']:.2f}
Ciências: {media_a['Ciências']:.2f}'''

print(mensagem)


Média das notas:
Matemática: 69.52
Ciências: 70.39


In [19]:
# Turma B
media_b = turma_b[['Matemática', 'Ciências']].mean()

mensagem = f'''Média das notas:
Matemática: {media_b['Matemática']:.2f}
Ciências: {media_b['Ciências']:.2f}'''

print(mensagem)

Média das notas:
Matemática: 70.89
Ciências: 71.56


In [20]:
# Turma C
media_c = turma_c[['Matemática', 'Ciências']].mean()

mensagem = f'''Média das notas:
Matemática: {media_c['Matemática']:.2f}
Ciências: {media_c['Ciências']:.2f}'''

print(mensagem)

Média das notas:
Matemática: 68.74
Ciências: 67.58


### b) Mediana:

In [26]:
# Turma A
mediana_a = turma_a[['Matemática', 'Ciências']].median()

mensagem = f'''Médiana das notas:
Matemática: {mediana_a['Matemática']:.2f}
Ciências: {mediana_a['Ciências']:.2f}'''

print(mensagem)

Médiana das notas:
Matemática: 68.00
Ciências: 72.00


In [27]:
# Turma B
mediana_b = turma_b[['Matemática', 'Ciências']].median()

mensagem = f'''Médiana das notas:
Matemática: {mediana_b['Matemática']:.2f}
Ciências: {mediana_b['Ciências']:.2f}'''

print(mensagem)

Médiana das notas:
Matemática: 71.00
Ciências: 72.00


In [28]:
# Turma C
mediana_c = turma_c[['Matemática', 'Ciências']].median()

mensagem = f'''Médiana das notas:
Matemática: {mediana_c['Matemática']:.2f}
Ciências: {mediana_c['Ciências']:.2f}'''

print(mensagem)

Médiana das notas:
Matemática: 74.00
Ciências: 67.00


### c) Moda:

In [35]:
# Turma A
moda_a = turma_a[['Matemática', 'Ciências']].mode().iloc[0]

mensagem = f'''Moda das notas:
Matemática: {moda_a['Matemática']}
Ciências: {moda_a['Ciências']}'''

print(mensagem)


Moda das notas:
Matemática: 94.0
Ciências: 71.0


In [36]:
# Turma B
moda_b = turma_b[['Matemática', 'Ciências']].mode().iloc[0]

mensagem = f'''Moda das notas:
Matemática: {moda_b['Matemática']}
Ciências: {moda_b['Ciências']}'''

print(mensagem)


Moda das notas:
Matemática: 54.0
Ciências: 58.0


In [37]:
# Turma C
moda_c = turma_c[['Matemática', 'Ciências']].mode().iloc[0]

mensagem = f'''Moda das notas:
Matemática: {moda_c['Matemática']}
Ciências: {moda_c['Ciências']}'''

print(mensagem)


Moda das notas:
Matemática: 78.0
Ciências: 42.0


### d) Amplitude:

In [43]:
# Turma A

amplitude = {
    'Matemática': turma_a['Matemática'].max() - turma_a['Matemática'].min(),
    'Ciências': turma_a['Ciências'].max() - turma_a['Ciências'].min()
}

mensagem = (
    f'''Amplitude das notas:
Matemática: {amplitude['Matemática']}
Ciências: {amplitude['Ciências']}'''
)

print(mensagem)

Amplitude das notas:
Matemática: 57
Ciências: 58


In [44]:
# Turma B

amplitude = {
    'Matemática': turma_b['Matemática'].max() - turma_b['Matemática'].min(),
    'Ciências': turma_b['Ciências'].max() - turma_b['Ciências'].min()
}

mensagem = (
    f'''Amplitude das notas:
Matemática: {amplitude['Matemática']}
Ciências: {amplitude['Ciências']}'''
)

print(mensagem)

Amplitude das notas:
Matemática: 53
Ciências: 58


In [45]:
# Turma C

amplitude = {
    'Matemática': turma_c['Matemática'].max() - turma_c['Matemática'].min(),
    'Ciências': turma_c['Ciências'].max() - turma_c['Ciências'].min()
}

mensagem = (
    f'''Amplitude das notas:
Matemática: {amplitude['Matemática']}
Ciências: {amplitude['Ciências']}'''
)

print(mensagem)

Amplitude das notas:
Matemática: 60
Ciências: 57


### e) Variância:

In [50]:
# Turma A

variancia = {
    'Matemática': turma_a['Matemática'].var(),
    'Ciências': turma_a['Ciências'].var()
}

mensagem = (
    f'''Variância das notas:
    Matemática: {variancia['Matemática']:.2f}
    Ciências: {variancia['Ciências']:.2f}
'''
)

print(mensagem)

Variância das notas:
    Matemática: 325.57
    Ciências: 346.00



In [51]:
# Turma B

variancia = {
    'Matemática': turma_b['Matemática'].var(),
    'Ciências': turma_b['Ciências'].var()
}

mensagem = (
    f'''Variância das notas:
    Matemática: {variancia['Matemática']:.2f}
    Ciências: {variancia['Ciências']:.2f}
'''
)

print(mensagem)

Variância das notas:
    Matemática: 234.10
    Ciências: 374.31



In [52]:
# Turma C

variancia = {
    'Matemática': turma_c['Matemática'].var(),
    'Ciências': turma_c['Ciências'].var()
}

mensagem = (
    f'''Variância das notas:
    Matemática: {variancia['Matemática']:.2f}
    Ciências: {variancia['Ciências']:.2f}
'''
)

print(mensagem)

Variância das notas:
    Matemática: 308.00
    Ciências: 347.32



### f) Desvio Padrão:

In [53]:
# Turma A

desvio_padrao = {
    'Matemática': turma_a['Matemática'].std(),
    'Ciências': turma_a['Ciências'].std()
}

mensagem = (
    f'''Desvio Padrão das notas:
    Matemática: {desvio_padrao['Matemática']:.2f}
    Ciências: {desvio_padrao['Ciências']:.2f}
'''
)

print(mensagem)

Desvio Padrão das notas:
    Matemática: 18.04
    Ciências: 18.60



In [54]:
# Turma B

desvio_padrao = {
    'Matemática': turma_b['Matemática'].std(),
    'Ciências': turma_b['Ciências'].std()
}

mensagem = (
    f'''Desvio Padrão das notas:
    Matemática: {desvio_padrao['Matemática']:.2f}
    Ciências: {desvio_padrao['Ciências']:.2f}
'''
)

print(mensagem)

Desvio Padrão das notas:
    Matemática: 15.30
    Ciências: 19.35



In [55]:
# Turma C

desvio_padrao = {
    'Matemática': turma_c['Matemática'].std(),
    'Ciências': turma_c['Ciências'].std()
}

mensagem = (
    f'''Desvio Padrão das notas:
    Matemática: {desvio_padrao['Matemática']:.2f}
    Ciências: {desvio_padrao['Ciências']:.2f}
'''
)

print(mensagem)

Desvio Padrão das notas:
    Matemática: 17.55
    Ciências: 18.64



### g)  Covariância e Correlação entre Matemática e Ciência:

In [56]:
# Turma A

cov_corr = {
    'Covariância': turma_a[['Matemática', 'Ciências']].cov().iloc[0, 1],
    'Correlação': turma_a[['Matemática', 'Ciências']].corr().iloc[0, 1]
}

mensagem = (
    f'''Covariância e Correlação das notas:
    Covariância (Matemática e Ciências): {cov_corr['Covariância']:.2f}
    Correlação (Matemática e Ciências): {cov_corr['Correlação']:.2f}
    '''
)

print(mensagem)

Covariância e Correlação das notas:
    Covariância (Matemática e Ciências): -68.65
    Correlação (Matemática e Ciências): -0.20
    


In [57]:
# Turma B

cov_corr = {
    'Covariância': turma_b[['Matemática', 'Ciências']].cov().iloc[0, 1],
    'Correlação': turma_b[['Matemática', 'Ciências']].corr().iloc[0, 1]
}

mensagem = (
    f'''Covariância e Correlação das notas:
    Covariância (Matemática e Ciências): {cov_corr['Covariância']:.2f}
    Correlação (Matemática e Ciências): {cov_corr['Correlação']:.2f}
    '''
)

print(mensagem)

Covariância e Correlação das notas:
    Covariância (Matemática e Ciências): -25.17
    Correlação (Matemática e Ciências): -0.09
    


In [58]:
# Turma C

cov_corr = {
    'Covariância': turma_c[['Matemática', 'Ciências']].cov().iloc[0, 1],
    'Correlação': turma_c[['Matemática', 'Ciências']].corr().iloc[0, 1]
}

mensagem = (
    f'''Covariância e Correlação das notas:
    Covariância (Matemática e Ciências): {cov_corr['Covariância']:.2f}
    Correlação (Matemática e Ciências): {cov_corr['Correlação']:.2f}
    '''
)

print(mensagem)

Covariância e Correlação das notas:
    Covariância (Matemática e Ciências): 77.05
    Correlação (Matemática e Ciências): 0.24
    


### Conclusões

1. **Com base nas medidas acima, qual turma teve o melhor desempenho geral em Matemática e Ciências?**
   - **Resposta**: A Turma B teve o melhor desempenho geral, com médias de 70.89 em Matemática e 71.56 em Ciências.

2. **Qual turma tem as notas mais consistentes em termos de desvio padrão?**
   - **Resposta**: A Turma B tem as notas mais consistentes, com desvios padrão de 15.30 em Matemática e 19.35 em Ciências, que são os menores entre as turmas.

3. **Existe uma correlação entre o desempenho em Matemática e Ciências para cada turma? Essa correlação é positiva ou negativa?**
   - **Resposta**:
     - **Turma A**: Correlação de -0.20 (negativa)
     - **Turma B**: Correlação de -0.09 (negativa)
     - **Turma C**: Correlação de 0.24 (positiva)

4. **Qual turma apresenta a maior amplitude de notas? O que isso nos diz sobre a desigualdade de desempenho dentro da turma?**
   - **Resposta**: A Turma C apresenta a maior amplitude de notas em Matemática (60) e uma amplitude de 57 em Ciências. Isso indica que a Turma C tem a maior desigualdade de desempenho, com uma grande variação nas notas dos alunos.
